R/export_portfolio.R

```diff
@@ -0,0 +1,74 @@
+#' @ export
+export_portfolio <- function(
+  portfolio_data,
+  group_data,
+  output_directory
+) {
+
+  logger::log_trace("cleaning and rearranging data prior to export")
+  output_cols <- c("isin", "market_value", "currency")
+  extra_cols <- setdiff(colnames(portfolio_data), output_cols)
+  if (length(extra_cols)) {
+    logger::log_warn(
+      "Extra columns detected in portfolio data: ",
+      extra_cols,
+      " Discarding."
+    )
+    warning("Extra columns detected in portfolio data. Discarding.")
+  }
+  missing_cols <- setdiff(output_cols, colnames(portfolio_data))
+  if (length(missing_cols)) {
+    logger::log_warn(
+      "Missing columns detected in portfolio data: ",
+      missing_cols,
+    )
+    stop("Missing columns detected in portfolio data.")
+  }
+
+  portfolio_data <- dplyr::select(
+    .data = portfolio_data,
+    dplyr::all_of(output_cols)
+  )
+
+  output_rows <- nrow(portfolio_data)
+
+  output_filename <- paste0(
+    uuid::UUIDgenerate(),
+    ".csv"
+  )
+
+  # Write the portfolio data to a file
+  output_filepath <- file.path(
+    output_directory,
+    output_filename
+  )
+
+  logger::log_trace("Writing portfolio data to file: ", output_filepath)
+  write.csv(
+    x = portfolio_data,
+    file = output_filepath,
+    row.names = FALSE,
+    fileEncoding = "UTF-8"
+  )
```

Comment:
I will say, the round-robin reading CSV then re-exporting CSV seems like it has potential for problems

Reply:
> the round-robin reading CSV then re-exporting CSV seems like it has potential for problems

That's possible, but my hope is that we catch the problems here, rather than when we try to actually run PACTA on a portfolio. The core concept of this workflow is that when a portfolio comes out the other side, it's in a format that PACTA can import cleanly

Reply:
You might want to consider some way of saving the original CSV to facilitate debugging when something wonky happens and the re-exported CSV doesn't match the original.